### PR TITLE
workspace-ified the netlink dependencies

### PR DESCRIPTION
**Pull request by anfredette**
Also added `features = ["tokio_socket"]` to `rtnetlink` in the bpfd `Cargo.toml` which became necessary after the title change.

### CODE REVIEW

**Comment by anfredette** [comment]
I'm not sure if this is the right solution.  We didn't see this build error before.  None of our related code changed, but it started failing at some point.

File: Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ resolver = "2"
 
 [patch.crates-io]
 aya = { git = "https://github.com/aya-rs/aya", branch = "main" }
+rtnetlink = { git = "https://github.com/rust-netlink/rtnetlink", branch = "main" }
```
Comment (astoycos) [inline Cargo.toml:15]: Is there an upstream issue number we can link here to make sure we know when to revert back to a release?

Comment (anfredette) [reply Cargo.toml]: It was a local cargo cache issues, so I deleted this change.

**Comment by anfredette** [comment]
It turned out that my original issue was a local one.  After clearing my cargo cache, the builds worked.  However, we still want to use workspaces for all the dependencies, so I left that change in this pr.

**Review by Billy99 — COMMENTED** [review]
LGTM

**Review by Billy99 — APPROVED** [review]

**Pull request merged**